<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Reporting toolbox for hap.py output • happyCompare</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="jquery.sticky-kit.min.js"></script><script src="pkgdown.js"></script><!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">happyCompare</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="/index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/simulations.html">Calibrating HDIs with simulated data</a>
    </li>
    <li>
      <a href="articles/stratified_counts.html">Analysing stratified counts</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    

    
    
<div class="contents">
<div id="happycompare" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#happycompare" class="anchor"></a>happyCompare</h1></div>

<p>happyCompary offers a set of functions to facilitate downstream analysis of variant calling performance outputs from <a href="https://github.com/Illumina/hap.py">hap.py</a>. It builds on top of <a href="https://github.com/Illumina/happyR">happyR</a> to support annotation of hap.py results (e.g. grouping) through metadata samplesheets, and provides methods for quick retrieval, statistical analysis and easy reporting of performance metrics.</p>
<div id="install" class="section level2">
<h2 class="hasAnchor">
<a href="#install" class="anchor"></a>Install</h2>
<p>Download <a href="https://git.illumina.com/Bioinformatics/happyCompare/archive/master.zip">zip</a>, extract and run:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">devtools::<span class="kw">install_local</span>(<span class="st">"path/to/happyCompare-master/"</span>)</code></pre></div>
</div>
<div id="usage" class="section level2">
<h2 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"happyCompare"</span>)
## Warning: Installed Rcpp (0.12.12) different from Rcpp used to build dplyr (0.12.11).
## Please reinstall dplyr to avoid random crashes or undefined behavior.

<span class="co"># loading demo data from a happyCompare samplesheet creates a happy_compare object...</span>
samplesheet_path &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"extdata/samplesheets"</span>, <span class="st">"pcrfree_vs_nano.readme.csv"</span>, <span class="dt">package =</span> <span class="st">"happyCompare"</span>)
happy_compare &lt;-<span class="st"> </span><span class="kw"><a href="reference/read_samplesheet.html">read_samplesheet</a></span>(samplesheet_path, <span class="dt">lazy =</span> <span class="ot">TRUE</span>)
<span class="kw">class</span>(happy_compare)
## [1] "happy_compare"

<span class="co"># ... that contains the following fields:</span>
<span class="co"># - samplesheet: the original samplesheet</span>
<span class="co"># - happy_results: a list of happy_result objects as defined in happyR</span>
<span class="co"># - ids: a vector of build ids</span>
<span class="kw">sapply</span>(happy_compare, class)
## $samplesheet
## [1] "tbl_df"     "tbl"        "data.frame"
## 
## $happy_results
## [1] "happy_result_list"
## 
## $ids
## [1] "character"

<span class="co"># hap.py results and samplesheet metadata can be accessed with extract_metrics(),</span>
<span class="co"># leaving them ready for downstream analysis</span>
e &lt;-<span class="st"> </span><span class="kw"><a href="reference/extract_metrics.html">extract_metrics</a></span>(happy_compare, <span class="dt">table =</span> <span class="st">"summary"</span>)
<span class="kw">class</span>(e)
## [1] "happy_summary" "tbl_df"        "tbl"           "data.frame"</code></pre></div>
</div>
<div id="example-visualisations" class="section level2">
<h2 class="hasAnchor">
<a href="#example-visualisations" class="anchor"></a>Example visualisations</h2>
<div id="summary-of-performance-metrics" class="section level3">
<h3 class="hasAnchor">
<a href="#summary-of-performance-metrics" class="anchor"></a>Summary of performance metrics</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># we can easily extract performance metrics and tabulate mean plus/minus SD </span>
<span class="co"># per group and variant type</span>
<span class="kw"><a href="reference/extract_metrics.html">extract_metrics</a></span>(happy_compare, <span class="dt">table =</span> <span class="st">"summary"</span>) %&gt;%<span class="st"> </span>
<span class="st">  </span><span class="kw">filter</span>(Filter ==<span class="st"> "PASS"</span>) %&gt;%<span class="st"> </span>
<span class="st">  </span><span class="kw"><a href="reference/hc_summarise_metrics.html">hc_summarise_metrics</a></span>(<span class="dt">df =</span> ., <span class="dt">group_cols =</span> <span class="kw">c</span>(<span class="st">"Group.Id"</span>, <span class="st">"Type"</span>)) %&gt;%<span class="st"> </span>
<span class="st">  </span>knitr::<span class="kw">kable</span>()</code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">Group.Id</th>
<th align="left">Type</th>
<th align="left">METRIC.F1_Score</th>
<th align="left">METRIC.Frac_NA</th>
<th align="left">METRIC.Precision</th>
<th align="left">METRIC.Recall</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Nano</td>
<td align="left">INDEL</td>
<td align="left">0.8725 ± 0.0024</td>
<td align="left">0.3428 ± 0.0078</td>
<td align="left">0.9315 ± 0.0036</td>
<td align="left">0.8205 ± 0.007</td>
</tr>
<tr class="even">
<td align="left">Nano</td>
<td align="left">SNP</td>
<td align="left">0.9707 ± 2e-04</td>
<td align="left">0.1425 ± 0.0016</td>
<td align="left">0.9963 ± 4e-04</td>
<td align="left">0.9465 ± 1e-04</td>
</tr>
<tr class="odd">
<td align="left">PCR-Free</td>
<td align="left">INDEL</td>
<td align="left">0.928 ± 1e-04</td>
<td align="left">0.3931 ± 0.0018</td>
<td align="left">0.9512 ± 2e-04</td>
<td align="left">0.9059 ± 1e-04</td>
</tr>
<tr class="even">
<td align="left">PCR-Free</td>
<td align="left">SNP</td>
<td align="left">0.9697 ± 4e-04</td>
<td align="left">0.1325 ± 0.0016</td>
<td align="left">0.9968 ± 1e-04</td>
<td align="left">0.9441 ± 8e-04</td>
</tr>
</tbody>
</table>
</div>
<div id="precision-recall-curves" class="section level3">
<h3 class="hasAnchor">
<a href="#precision-recall-curves" class="anchor"></a>Precision-recall curves</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># similarly, we can extract roc metrics and plot a precision-recall curve for PASS INDEL</span>
<span class="kw"><a href="reference/extract_metrics.html">extract_metrics</a></span>(happy_compare, <span class="dt">table =</span> <span class="st">"pr.all"</span>) %&gt;%<span class="st"> </span>
<span class="st">  </span><span class="kw"><a href="reference/hc_plot_roc.html">hc_plot_roc</a></span>(<span class="dt">happy_roc =</span> ., <span class="dt">type =</span> <span class="st">"INDEL"</span>, <span class="dt">filter =</span> <span class="st">"PASS"</span>)</code></pre></div>
<p><img src="examples/README-unnamed-chunk-5-1.png" width="480"></p>
</div>
<div id="stratified-counts" class="section level3">
<h3 class="hasAnchor">
<a href="#stratified-counts" class="anchor"></a>Stratified counts</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># finally, we can extract stratified counts and estimate highest density intervals </span>
<span class="co"># for recall in level 0 subsets...</span>
hdi &lt;-<span class="st"> </span><span class="kw"><a href="reference/extract_metrics.html">extract_metrics</a></span>(happy_compare, <span class="dt">table =</span> <span class="st">"extended"</span>) %&gt;%<span class="st"> </span>
<span class="st">  </span><span class="kw">filter</span>(Subtype ==<span class="st"> "*"</span>, Filter ==<span class="st"> "PASS"</span>, Subset.Level ==<span class="st"> </span><span class="dv">0</span>, 
         Subset %in%<span class="st"> </span><span class="kw">c</span>(<span class="st">"high.at"</span>, <span class="st">"high.gc"</span>)) %&gt;%<span class="st"> </span>
<span class="st">  </span><span class="kw"><a href="reference/estimate_hdi.html">estimate_hdi</a></span>(<span class="dt">successes_col =</span> <span class="st">"TRUTH.TP"</span>, <span class="dt">totals_col =</span> <span class="st">"TRUTH.TOTAL"</span>, 
               <span class="dt">group_cols =</span> <span class="kw">c</span>(<span class="st">"Group.Id"</span>, <span class="st">"Subset"</span>, <span class="st">"Type"</span>), <span class="dt">aggregate_only =</span> <span class="ot">FALSE</span>)

<span class="co"># ... and generate custom plots with ggplot2</span>
hdi %&gt;%<span class="st"> </span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">Subset =</span> <span class="kw">factor</span>(Subset, <span class="dt">levels =</span> <span class="kw">rev</span>(<span class="kw">unique</span>(Subset)))) %&gt;%<span class="st"> </span>
<span class="st">  </span><span class="kw">filter</span>(replicate_id ==<span class="st"> ".aggregate"</span>) %&gt;%<span class="st"> </span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> estimated_p, <span class="dt">y =</span> Subset, <span class="dt">group =</span> Subset)) +
<span class="st">    </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">color =</span> Group.Id), <span class="dt">size =</span> <span class="dv">2</span>) +
<span class="st">    </span><span class="kw">geom_errorbarh</span>(<span class="kw">aes</span>(<span class="dt">xmin =</span> lower, <span class="dt">xmax =</span> upper, <span class="dt">color =</span> Group.Id), <span class="dt">height =</span> <span class="fl">0.4</span>) +
<span class="st">    </span><span class="kw">facet_grid</span>(. ~<span class="st"> </span>Type) +
<span class="st">    </span><span class="kw">scale_colour_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">"#E69F00"</span>, <span class="st">"#56B4E9"</span>)) +
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">"bottom"</span>) +
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">"Recall estimates across L0 subsets"</span>) +
<span class="st">    </span><span class="kw">xlab</span>(<span class="st">"Recall"</span>) +
<span class="st">    </span><span class="kw">ylab</span>(<span class="st">""</span>) +
<span class="st">    </span><span class="kw">xlim</span>(<span class="fl">0.7</span>, <span class="dv">1</span>)</code></pre></div>
<p><img src="examples/README-unnamed-chunk-6-1.png" width="672"></p>
</div>
</div>
<div id="system-requirements" class="section level2">
<h2 class="hasAnchor">
<a href="#system-requirements" class="anchor"></a>System requirements</h2>
<div id="hardware-os-and-r" class="section level3">
<h3 class="hasAnchor">
<a href="#hardware-os-and-r" class="anchor"></a>Hardware, OS and R</h3>
<p>Development and testing for <code>happyCompare</code> have been done using R 3.3.3 on a Centos 6.9 machine, with 2 processor cores and 8GB of RAM. Alternative systems are also supported as documented in the official <a href="https://cran.r-project.org/doc/manuals/r-release/R-admin.html">R installation guide</a>.</p>
</div>
<div id="r-session-info" class="section level3">
<h3 class="hasAnchor">
<a href="#r-session-info" class="anchor"></a>R session info</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">devtools::<span class="kw">session_info</span>()</code></pre></div>
<pre><code>## Session info -------------------------------------------------------------</code></pre>
<pre><code>##  setting  value                       
##  version  R version 3.3.3 (2017-03-06)
##  system   x86_64, linux-gnu           
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  tz       &lt;NA&gt;                        
##  date     2017-08-17</code></pre>
<pre><code>## Packages -----------------------------------------------------------------</code></pre>
<pre><code>##  package      * version    date       source                         
##  assertthat     0.2.0      2017-04-11 CRAN (R 3.3.3)                 
##  backports      1.1.0      2017-05-22 CRAN (R 3.3.3)                 
##  base         * 3.3.3      2017-03-08 local                          
##  bindr          0.1        2016-11-13 cran (@0.1)                    
##  bindrcpp     * 0.2        2017-06-17 cran (@0.2)                    
##  broom          0.4.2      2017-02-13 CRAN (R 3.3.3)                 
##  cellranger     1.1.0      2016-07-27 CRAN (R 3.3.3)                 
##  colorspace     1.3-2      2016-12-14 CRAN (R 3.3.3)                 
##  commonmark     1.2        2017-03-01 CRAN (R 3.3.3)                 
##  crayon         1.3.2      2016-06-28 CRAN (R 3.3.3)                 
##  datasets     * 3.3.3      2017-03-08 local                          
##  desc           1.1.0      2017-01-27 CRAN (R 3.3.3)                 
##  devtools       1.13.1     2017-05-13 CRAN (R 3.3.3)                 
##  digest         0.6.12     2017-01-27 CRAN (R 3.3.3)                 
##  dplyr        * 0.7.0      2017-06-09 cran (@0.7.0)                  
##  evaluate       0.10.1     2017-06-24 cran (@0.10.1)                 
##  forcats        0.2.0      2017-01-23 CRAN (R 3.3.3)                 
##  foreign        0.8-67     2016-09-13 CRAN (R 3.3.3)                 
##  ggplot2      * 2.2.1      2016-12-30 CRAN (R 3.3.3)                 
##  glue           1.1.1      2017-06-21 cran (@1.1.1)                  
##  graphics     * 3.3.3      2017-03-08 local                          
##  grDevices    * 3.3.3      2017-03-08 local                          
##  grid           3.3.3      2017-03-08 local                          
##  gtable         0.2.0      2016-02-26 CRAN (R 3.3.3)                 
##  happyCompare * 1.0.0      2017-08-16 local                          
##  happyR         0.1.0      2017-08-15 local (@5b26996)               
##  haven          1.0.0      2016-09-23 CRAN (R 3.3.3)                 
##  HDInterval     0.1.3      2016-05-13 CRAN (R 3.3.3)                 
##  highr          0.6        2016-05-09 CRAN (R 3.3.3)                 
##  hms            0.3        2016-11-22 CRAN (R 3.3.3)                 
##  htmltools      0.3.6      2017-04-28 CRAN (R 3.3.3)                 
##  httr           1.2.1      2016-07-03 CRAN (R 3.3.3)                 
##  jsonlite       1.5        2017-06-01 CRAN (R 3.3.3)                 
##  knitr          1.16       2017-05-18 CRAN (R 3.3.3)                 
##  labeling       0.3        2014-08-23 CRAN (R 3.3.3)                 
##  lattice        0.20-34    2016-09-06 CRAN (R 3.3.3)                 
##  lazyeval       0.2.0      2016-06-12 CRAN (R 3.3.3)                 
##  lubridate      1.6.0      2016-09-13 CRAN (R 3.3.3)                 
##  magrittr       1.5        2014-11-22 CRAN (R 3.3.3)                 
##  MASS           7.3-45     2016-04-21 CRAN (R 3.3.3)                 
##  memoise        1.1.0      2017-04-21 CRAN (R 3.3.3)                 
##  methods      * 3.3.3      2017-03-08 local                          
##  mnormt         1.5-5      2016-10-15 CRAN (R 3.3.3)                 
##  modelr         0.1.0      2016-08-31 CRAN (R 3.3.3)                 
##  munsell        0.4.3      2016-02-13 CRAN (R 3.3.3)                 
##  nlme           3.1-131    2017-02-06 CRAN (R 3.3.3)                 
##  parallel       3.3.3      2017-03-08 local                          
##  pkgconfig      2.0.1      2017-03-21 cran (@2.0.1)                  
##  pkgdown        0.1.0.9000 2017-07-06 Github (hadley/pkgdown@7cde4c8)
##  plyr           1.8.4      2016-06-08 CRAN (R 3.3.3)                 
##  psych          1.7.5      2017-05-03 CRAN (R 3.3.3)                 
##  purrr        * 0.2.2.2    2017-05-11 CRAN (R 3.3.3)                 
##  R6             2.2.2      2017-06-17 cran (@2.2.2)                  
##  Rcpp           0.12.12    2017-07-15 cran (@0.12.12)                
##  readr        * 1.1.1      2017-05-16 CRAN (R 3.3.3)                 
##  readxl         1.0.0      2017-04-18 CRAN (R 3.3.3)                 
##  reshape2       1.4.2      2016-10-22 CRAN (R 3.3.3)                 
##  rlang          0.1.2.9000 2017-08-15 Github (hadley/rlang@0e62148)  
##  rmarkdown      1.6        2017-06-15 cran (@1.6)                    
##  roxygen2       6.0.1      2017-02-06 CRAN (R 3.3.3)                 
##  rprojroot      1.2        2017-01-16 CRAN (R 3.3.3)                 
##  rvest          0.3.2      2016-06-17 CRAN (R 3.3.3)                 
##  scales         0.4.1      2016-11-09 CRAN (R 3.3.3)                 
##  stats        * 3.3.3      2017-03-08 local                          
##  stringi        1.1.5      2017-04-07 CRAN (R 3.3.3)                 
##  stringr        1.2.0      2017-02-18 CRAN (R 3.3.3)                 
##  tibble       * 1.3.3      2017-05-28 CRAN (R 3.3.3)                 
##  tidyr        * 0.6.3      2017-05-15 CRAN (R 3.3.3)                 
##  tidyverse    * 1.1.1      2017-01-27 CRAN (R 3.3.3)                 
##  tools          3.3.3      2017-03-08 local                          
##  utils        * 3.3.3      2017-03-08 local                          
##  withr          1.0.2      2016-06-20 CRAN (R 3.3.3)                 
##  xml2           1.1.1      2017-01-24 CRAN (R 3.3.3)                 
##  yaml           2.1.14     2016-11-12 CRAN (R 3.3.3)</code></pre>
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2 class="hasAnchor">
<a href="#sidebar" class="anchor"></a>License</h2>
<p><a href="https://opensource.org/licenses/BSD-3-Clause">BSD</a></p>
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Mar Gonzalez-Porta <br><small class="roles"> Author, maintainer </small> </li>
<li>Ben Moore <br><small class="roles"> Author </small> </li>
</ul>
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://travis-ci.org/Illumina/happyCompare"><img src="https://travis-ci.org/Illumina/happyCompare.svg?branch=master" alt="Build Status"></a></li>
<li><a href="https://codecov.io/github/Illumina/happyCompare?branch=master"><img src="https://codecov.io/github/Illumina/happyCompare/coverage.svg?branch=master" alt="Coverage Status"></a></li>
</ul>
</div>

</div>


      <footer><div class="copyright">
  <p>Developed by Mar Gonzalez-Porta, Ben Moore.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
